### PR TITLE
ADR 010, documentation about db transactions and locks

### PR DESCRIPTION
**Pull request by opqdonut**
for #2101

### CODE REVIEW

File: docs/architecture/010-transactions.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+## Strict ordering of commands via locking
+
+Our application commands are processed roughly in the following way
+(implementation: `rems.api.services.command/command!`):
+
+1. get all events for application from the `application_event` table
+2. compute application state
+3. run command handler (given command and application state)
+4. if command handler succeeds, append new events to `application_event`
+5. run process managers on new events, generating e.g. rows in the outbox table
```
[the remark this answers is not in the document]
Comment (opqdonut) [reply docs/architecture/010-transactions.md:35]: done

File: docs/architecture/010-transactions.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+read-only to guarantee that our GETs are pure.
+
+[PostgreSQL docs]: https://www.postgresql.org/docs/9.6/transaction-iso.html
+
+## Strict ordering of commands via locking
+
+Our application commands are processed roughly in the following way
+(implementation: `rems.api.services.command/command!`):
+
+1. get all events for application from the `application_event` table
```
[the remark this answers is not in the document]
Comment (opqdonut) [reply docs/architecture/010-transactions.md:31]: It looks like that cache has been removed and we only have the `all-applications-cache` which is used for mass GET requests but not in `get-unrestricted-application` which `rems.api.services.command` uses

[the remark this answers is not in the document]
Comment (Macroz) [reply docs/architecture/010-transactions.md:31]: Well the all applications cache does exist though this code path doesn't use it directly. Its update is done somewhere?

File: docs/architecture/010-transactions.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+2. compute application state
+3. run command handler (given command and application state)
+4. if command handler succeeds, append new events to `application_event`
+5. run process managers on new events, which can result in
+   - new commands (e.g. from `approver-bot`)
+   - new rows in db (e.g. `outbox` or `blacklist` rows)
```
Comment (Macroz) [inline docs/architecture/010-transactions.md:37]: >>>
These commands are processed immediately, in the same thread.

Comment (opqdonut) [reply docs/architecture/010-transactions.md:37]: done

File: docs/architecture/010-transactions.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (opqdonut) [reply docs/architecture/010-transactions.md:31]: I added a comment

**Comment by opqdonut** [comment]
related: #2100

**Pull request merged**